src/linker/Linker/Driver.cs

```diff
@@ -49,11 +50,6 @@ public partial class Driver : IDisposable
 #endif
 
 		public static int Main (string[] args)
-		{
-			return Execute (args);
-		}
-
-		public static int Execute (string[] args, ILogger customLogger = null)
 		{
 			if (args.Length == 0) {
 				Console.Error.WriteLine ("No parameters specified");
```

Comment:
Should we display `Usage()` in this case?

Reply:
Nit: Should this be the other way round? First print out the error and then usage?

Reply:
I think usage first makes sense so that the error isn't hidden by the large amount of usage output. If we want to avoid printing usage, maybe we should instead support standard `-h` and `--help` options? Right now it takes some guessing to stumble on `-?` or `-help` for someone not familiar with the tool.

Reply:
Sure - it was jut a nit.

Re printing out usage at all - I think we should print out something. Currently running linker without any command line doesn't provide any actionable message. "No parameters specified" -> "Well yeah, I know, I just did that, thanks for telling me... but what am I supposed to do instead?".

I can see the argument of not printing Usage when there's an error in the command line, but empty command line should be a special case. I personally would like to see Usage, but if not that, then I agree with @sbomer that we should at the very least print out how to get to the usage (although it feels like unnecessary two step process).

Reply:
I think the situation is improved with `-h` and `--help`. I opened https://github.com/mono/linker/issues/1353 where we can discuss any follow-ups.